ykcapi/src/lib.rs

```diff
@@ -67,18 +67,59 @@ pub extern "C" fn __ykrt_control_point(
     // Stackmap id for the control point.
     smid: u64,
 ) {
-    // FIXME: We could get rid of this entire function if we pass the frame's base pointer into the
-    // control point from the interpreter.
-    std::arch::naked_asm!(
-        // Pass the interpreter frame's base pointer via the 4th argument register.
-        "sub rsp, 8",   // Alignment
-        "mov rcx, rbp", // Pass interpreter frame's base pointer via 4th argument register.
-        "call __ykrt_control_point_real",
-        "add rsp, 8",
-        "ret",
-    );
+    #[cfg(not(swt_modclone))]
+    {
+        // FIXME: We could get rid of this entire function if we pass the frame's base pointer into the
+        // control point from the interpreter.
+        std::arch::naked_asm!(
+            // Pass the interpreter frame's base pointer via the 4th argument register.
+            "sub rsp, 8",   // Alignment
+            "mov rcx, rbp", // Pass interpreter frame's base pointer via 4th argument register.
+            "call __ykrt_control_point_real",
+            "add rsp, 8",
+            "ret",
+        );
+    }
+    #[cfg(swt_modclone)]
+    {
+        // FIXME: Adapt multi version swt control point transition.
```

Comment:
I remember you asking if we can use the same trick here as in https://github.com/ykjit/yk/pull/1813. I changed my mind about it. We might not be able to get rid of the callee-saved registers here, but we should be able to remove the caller-saved registers by reading them from the extra locations instead. We don't need to do this here, but it might be worth investigating in a follow up PR. Maybe the callee-saved registers can also be removed by returning into the transitioning routine, as we do with traces we want to execute, but I'm not sure yet if/how that would work.